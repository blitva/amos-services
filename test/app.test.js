import React from 'react';
import ReactDOM from 'react-dom';
import { mount, shallow } from 'enzyme';
import App from '../client/src/components/app.jsx';
import ShallowRenderer from 'react-test-renderer/shallow';
import renderer from 'react-test-renderer';
import toJson from 'enzyme-to-json';
import ItemHeading from '../client/src/components/itemHeading.jsx';
import ItemPriceDetails from '../client/src/components/itemPriceDetails.jsx';
import ItemConfiguration from '../client/src/components/itemConfiguration.jsx';
import ItemColorOptions from '../client/src/components/itemColorOptions.jsx';
import ItemDescription from '../client/src/components/itemDescription.jsx';


//test whether each component renders
describe('Components Should Render', () => {

  test('App component should render correctly onto the page', () => {
    const appWrapper = shallow(<App />);
    expect(toJson(appWrapper)).toMatchSnapshot();
  });

  test('ItemHeading component should render correctly', () => {
    const itemHeadingWrapper = shallow(<ItemHeading/>);
    expect(toJson(itemHeadingWrapper)).toMatchSnapshot();
  });

  test('ItemPriceDetails component should render correctly', () => {
    const itemPriceWrapper = shallow(<ItemPriceDetails />);
    expect(toJson(itemPriceWrapper)).toMatchSnapshot();
  });

  test('ItemColorOptions component should render correctly', () => {
    const itemColors = shallow(<ItemColorOptions />);
    expect(toJson(itemColors)).toMatchSnapshot();
  });

  test('ItemConfiguration component should render correctly', () => {
    const itemConfiguration = shallow(<ItemConfiguration />);
    expect(toJson(itemConfiguration)).toMatchSnapshot();
  });

  test('ItemDescription component should render correctly', () => {
    const itemDescription = shallow(<ItemDescription />);
    expect(toJson(itemDescription)).toMatchSnapshot();
  });
});