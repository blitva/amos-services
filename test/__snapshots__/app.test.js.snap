// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Should Render App component should render correctly onto the page 1`] = `
<h1>
  HELLO SERVICE IS RUNNING!
</h1>
`;

exports[`Components Should Render App component should render correctly onto the page 2`] = `
<div>
  <div>
    title
  </div>
  <div>
    ratings
  </div>
  <div>
    questions answered
  </div>
</div>
`;

exports[`Components Should Render ItemColorOptions component should render correctly 1`] = `<div />`;

exports[`Components Should Render ItemConfiguration component should render correctly 1`] = `
<div>
  Item Configuration
</div>
`;

exports[`Components Should Render ItemDescription component should render correctly 1`] = `
<div>
  Item Description
</div>
`;

exports[`Components Should Render ItemHeading component should render correctly 1`] = `
<div>
  <div>
    title
  </div>
  <div>
    ratings
  </div>
  <div>
    questions answered
  </div>
</div>
`;

exports[`Components Should Render ItemPriceDetails component should render correctly 1`] = `
<div>
  Item price Details
</div>
`;
